apps/emqx_authz/test/emqx_authz_jwt_SUITE.erl

```diff
@@ -0,0 +1,312 @@
+        {deny, <<"testsub1/clientid">>},
+        {deny, <<"testsub4/foobar">>},
+
+        {deny, <<"testall1/username">>},
+        {allow, <<"testall1/${username}">>},
+        {allow, <<"testall2/clientid">>},
+        {allow, <<"testall3/foobar">>},
+        {allow, <<"testall3/+/foobar">>},
+        {allow, <<"testall3/#">>},
+
+        {deny, <<"testall2/username">>},
+        {deny, <<"testall1/clientid">>},
+        {deny, <<"testall4/foobar">>}
+    ],
+
+    lists:foreach(
+        fun
+            ({allow, Topic}) ->
+                ?assertMatch(
+                    {ok, #{}, [0]},
+                    emqtt:subscribe(C, Topic, 0)
+                );
+            ({deny, Topic}) ->
+                ?assertMatch(
+                    {ok, #{}, [?RC_NOT_AUTHORIZED]},
+                    emqtt:subscribe(C, Topic, 0)
+                )
+        end,
+        Cases
+    ),
+
+    ok = emqtt:disconnect(C).
+
+t_check_pub(_Config) ->
+    Payload = #{
+        <<"username">> => <<"username">>,
+        <<"acl">> => #{<<"sub">> => [<<"a/b">>]},
+        <<"exp">> => erlang:system_time(second) + 10
+    },
+
+    JWT = generate_jws(Payload),
+
+    {ok, C} = emqtt:start_link(
+        [
+            {clean_start, true},
+            {proto_ver, v5},
+            {clientid, <<"clientid">>},
+            {username, <<"username">>},
+            {password, JWT}
+        ]
+    ),
+    {ok, _} = emqtt:connect(C),
+
+    ok = emqtt:publish(C, <<"a/b">>, <<"hi">>, 0),
+
+    receive
+        {publish, #{topic := <<"a/b">>}} ->
+            ?assert(false, "Publish to `a/b` should not be allowed")
+    after 100 -> ok
+    end,
+
+    ok = emqtt:disconnect(C).
+
+t_check_no_recs(_Config) ->
+    Payload = #{
+        <<"username">> => <<"username">>,
+        <<"acl">> => #{},
+        <<"exp">> => erlang:system_time(second) + 10
+    },
+
+    JWT = generate_jws(Payload),
+
+    {ok, C} = emqtt:start_link(
+        [
+            {clean_start, true},
+            {proto_ver, v5},
+            {clientid, <<"clientid">>},
+            {username, <<"username">>},
+            {password, JWT}
+        ]
+    ),
+    {ok, _} = emqtt:connect(C),
+
+    ?assertMatch(
+        {ok, #{}, [?RC_NOT_AUTHORIZED]},
+        emqtt:subscribe(C, <<"a/b">>, 0)
+    ),
+
+    ok = emqtt:disconnect(C).
+
+t_check_no_acl_claim(_Config) ->
+    Payload = #{
+        <<"username">> => <<"username">>,
+        <<"exp">> => erlang:system_time(second) + 10
+    },
+
+    JWT = generate_jws(Payload),
+
+    {ok, C} = emqtt:start_link(
+        [
+            {clean_start, true},
+            {proto_ver, v5},
+            {clientid, <<"clientid">>},
+            {username, <<"username">>},
+            {password, JWT}
+        ]
+    ),
+    {ok, _} = emqtt:connect(C),
+
+    ?assertMatch(
+        {ok, #{}, [?RC_NOT_AUTHORIZED]},
+        emqtt:subscribe(C, <<"a/b">>, 0)
+    ),
+
+    ok = emqtt:disconnect(C).
+
+t_check_expire(_Config) ->
```

Comment:
Is it worth to also test `nbf` and `iat`?

Reply:
with invalid `nbf` and `iat` a user can't authenticate... so seems that we should just stop authorizing after token expire.